src/Sparrow/Json/JsonContextPoolBase.cs

```diff
@@ -426,10 +278,30 @@ public void LowMemoryOver()
         }
     }
 
-    public static class JsonContextPoolWorkStealing
+    internal class JsonContextPoolHelpers
     {
-        [ThreadStatic]
-        public static bool AvoidForCurrentThread;
+        internal const string LIBC_6 = "libc";
+
+        [DllImport(LIBC_6, EntryPoint = "sched_getcpu", CallingConvention = CallingConvention.Cdecl, SetLastError = false)]
+        private static extern int sched_getcpu();
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        private static extern int GetCurrentProcessorNumber();
+
+        public static int CurrentProcessNumber
+        {
+            get
+            {
+                if (PlatformDetails.RunningOnPosix)
```

Comment:
Need to measure whatever we have to cache that, see the ProcessorIdCache here:
https://github.com/dotnet/runtime/blob/master/src/coreclr/src/System.Private.CoreLib/src/System/Threading/ProcessorIdCache.cs